ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/methods/EngineNewPayloadV1.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ethereum.executionclient.methods;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.ethereum.executionclient.ExecutionEngineClient;
+import tech.pegasys.teku.ethereum.executionclient.response.ResponseUnwrapper;
+import tech.pegasys.teku.ethereum.executionclient.schema.ExecutionPayloadV1;
+import tech.pegasys.teku.ethereum.executionclient.schema.PayloadStatusV1;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.spec.datastructures.execution.ExecutionPayload;
+import tech.pegasys.teku.spec.executionlayer.PayloadStatus;
+
+public class EngineNewPayloadV1 extends AbstractEngineJsonRpcMethod<PayloadStatus> {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  public EngineNewPayloadV1(final ExecutionEngineClient executionEngineClient) {
+    super(executionEngineClient);
+  }
+
+  @Override
+  public String getName() {
+    return "engine_newPayload";
+  }
+
+  @Override
+  public int getVersion() {
+    return 1;
+  }
+
+  @Override
+  public SafeFuture<PayloadStatus> execute(final JsonRpcRequestParams params) {
+    final ExecutionPayload executionPayload =
+        params.getRequiredParameter(0, ExecutionPayload.class);
+
+    LOG.trace("calling engineNewPayloadV1(executionPayload={})", executionPayload);
```

Comment:
one thing we could potentially do here is add a check where if the spec exceeds bellatrix, we fail with a meaningful message...

It might be a good pattern to follow, so if v2 is called with deneb we can also fail fast - thoughts?

Reply:
I like this idea. I'll do it later after we add all methods in. Basically, I am avoiding anything that changes the current functionality so we can still use all test as they are to ensure correctness.